### PR TITLE
Fix: Pass correct language ID to getAttributesName instead of hardcoded 1

### LINKED ISSUE #36458: Hardcoded Language ID in getAttributesName Call Leads to Incorrect Attribute Names
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Description of the Bug:

In the `AdminProductWrapper.php` file, while fetching the specific prices list for a product,  the method `getAttributesName` is called with a hardcoded parameter 1 for the language ID. 

https://github.com/PrestaShop/PrestaShop/blob/8.1.x/src/Adapter/Product/AdminProductWrapper.php#L481

This implementation assumes that language id 1 is always appropriate for fetching attribute names, which is not always the case. In scenarios where the application is being used in a multilingual context, or when the user's preferred language is different from the default language, this approach leads to incorrect attribute names being fetched and displayed. This can cause confusion for the users and potentially lead to incorrect data handling or display issues in the application. The bug essentially limits the application's flexibility and usability in multilingual environments.

### Expected behavior

When i change the language of the page, the name of the attribute should be displayed in the good language.

### Steps to reproduce

1. Go to Product (with declianison) page in BO
2. Click Price tab
3. In the specific price list, The name will not be displayed well if the default language was different one. because it always fetches the name with id lang 1.

### PrestaShop version(s) where the bug happened

8.1.x

### PHP version(s) where the bug happened

_No response_

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by mouleeg**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | The language id param is set to 1 by default. This causes an issue in fetching the good name when changing the language in BO -> product page -> Price tab -> Specific price list. Using the language id in context should fix the issue.  
| Type?             | bug fix 
| Category?         | BO 
| BC breaks?        |  no
| Deprecations?     | no
| How to test?      |  BO -> product page -> Prix tab -> Specific price list. Create a specific price for a declinaison.  In the list, the name of declinaison will be displayed well with this fix. Before that only the name in english language will be displayed.
| UI Tests          | https://github.com/mouleeg/ga.tests.ui.pr/actions/workflows/pr_test.yml https://github.com/florine2623/testing_pr/actions/runs/9740649087
| Fixed issue or discussion?     | #36458 
| Related PRs       | 
| Sponsor company   |

### CODE REVIEW

**Comment by ps-jarvis** [comment]
Hi, thanks for this contribution!

Please consider opening an issue before submitting a Pull Request:
* If it's a bug fix, it helps maintainers verify that the bug is effectively due to a defect in the code, and that it hasn't been fixed already.
* It can help trigger a discussion about the best implementation path before a single line of code is written.
* It may lead the Core Product team to mark that issue as a priority, further attracting the maintainers' attention.

(Note: this is an automated message, but answering it will reach a real human)

**Comment by ps-jarvis** [comment]
Hello @mouleeg!

This is your first pull request on [PrestaShop](https://github.com/PrestaShop/PrestaShop/) repository of the PrestaShop project.

Thank you, and welcome to this Open Source community!

**Review by Hlavtox — APPROVED** [review]
LGTM, good catch 👍

**Review by boherm — APPROVED** [review]

**Review by PululuK — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @mouleeg ,

Thanks for the PR !

Tested for new product page and old product page.
Tested with one/multiple combinations.

New : 

https://github.com/PrestaShop/PrestaShop/assets/16019289/464c62f5-05cb-4d7d-982d-9fedc9ef2a17

Old :

https://github.com/PrestaShop/PrestaShop/assets/16019289/7ab94ae3-ce54-47e7-aa46-d634a77dc981

LGTM, it is QA ✅

**Comment by florine2623** [comment]
Please, wait before merging.

@PrestaShop/qa-automation 
Auto tests are failing, is it related to this PR ?

**Review by Hlavtox — DISMISSED** [review]
Block until QA reply to avoid accidental merge.

Dismiss this and merge it after. :-)

**Comment by florine2623** [comment]
Tests are green now ^^

**Comment by Hlavtox** [comment]
Thanks @mouleeg and @florine2623! :-)
The fix will be available in 8.1.8.

**Pull request merged**
